python/dataset.cpp

```diff
@@ -134,6 +135,33 @@ void init_dataset(py::module &m) {
 
   bind_data_properties(dataProxy);
 
+  m.def("histogram",
+        [](const DataConstProxy &ds, const Variable &bins) {
+          return core::histogram(ds, bins);
+        },
+        py::call_guard<py::gil_scoped_release>(),
+        "Returns a new Variable with values in bins for for sparse dims");
+
+  m.def("histogram",
+        [](const DataConstProxy &ds, const VariableConstProxy &bins) {
+          return core::histogram(ds, bins);
+        },
+        py::call_guard<py::gil_scoped_release>(),
+        "Returns a new Variabble with values in bins for sparse dims");
+
+  m.def("histogram",
+        [](const Dataset &ds, const Variable &bins) {
+          return core::histogram(ds, bins);
+        },
+        py::call_guard<py::gil_scoped_release>(),
+        "Returns a new Dataset with histograms for sparse dims");
+
+  m.def("histogram",
+        [](const Dataset &ds, const VariableConstProxy &bins) {
+          return core::histogram(ds, bins);
+        },
+        py::call_guard<py::gil_scoped_release>(),
+        "Returns a new Dataset with histograms for sparse dims");
```

Comment:
Can you make sure this is actually calling the overloads for `VariableConstProxy`? We have implicit conversion, so there is a risk that pybind11 is just calling the first overload, using costly conversion.